resources/deployments/device_deployment.go

```diff
@@ -37,6 +37,17 @@ const (
 	DeviceDeploymentStatusDecommissioned = "decommissioned"
 )
 
+// DeviceDeploymentStatus is a helper type for reporting status changes through
+// the layers
+type DeviceDeploymentStatus struct {
+	// status reported by device
+	Status string `valid:"required"`
+	// substate reported by device
+	SubState *string
```

Comment:
any length limit?

Reply:
Should be fixed now, `SubState` is limited to 200 characters.